### PR TITLE
feat: allow memos in pwa

### PR DESCRIPTION
**Pull request by UncleSamtoshi**
- create dynamic manifests using the api and allowing for bypass of `/setuppwa` page
- allow memo to be passed into query string

### CODE REVIEW

**Review by Extheoisah — COMMENTED** [review]
Just two comments:
1.) I think you need to do a rebase to pull in the new changes from main
2.) I can't find an input component where the user enters the memo details or do they just edit it in the URL?

File: pages/_app.tsx [hunk 1]
```diff
@@ -25,7 +35,13 @@ export default function Layout({
...
         <link rel="apple-touch-icon" href="/APPLE-ICON.png" />
         <link rel="icon" type="image/png" href="/APPLE-ICON.png" />
-        <link rel="manifest" href="/manifest.json" id="manifest" />
+        {username ? (
+          <link
+            rel="manifest"
+            href={`/api/${username}/manifest?${manifestParms.toString()}`}
+            id={`${username}/manifest`}
+          />
+        ) : null}
```
Comment (Extheoisah) [inline pages/_app.tsx:44]: I think it's better to return a default manifest in the ternary.

File: pages/_app.tsx [hunk 2]
```diff
@@ -15,6 +16,15 @@ export default function Layout({
   Component: NextPage
   pageProps: Record<string, unknown>
 }) {
+  const {
+    query: { username, memo },
+  } = useRouter()
+
+  const manifestParms = new URLSearchParams()
```
[the remark this answers is not in the document]
Comment (UncleSamtoshi) [reply pages/_app.tsx:23]: Will fix this.

File: pages/_app.tsx (hunk 1 above)
Comment (UncleSamtoshi) [reply pages/_app.tsx:44]: I was able to resolve the issue with the icons not showing up on chrome. However if I include the default manifest the browsers will default to using it, and it is not possible to override it.

**Review by Extheoisah — DISMISSED** [review]
LGTM

**Review by Extheoisah — DISMISSED** [review]

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**